src/test/java/use_cases/PlayerGameInteractorTest.java

```diff
@@ -0,0 +1,221 @@
+            // Card was the played Card.
+            assertTrue(response.getHasWinner());
+            assertEquals(0, p1.getNumCards());
+            HumanPlayer humanP1 = (HumanPlayer) p1;
+            assertEquals(1, humanP1.getStats()[0]);
+            assertEquals(0, humanP1.getStats()[1]);
+            HumanPlayer humanP3 = (HumanPlayer) p3;
+            assertEquals(1, humanP3.getStats()[1]);
+            assertEquals(0, humanP3.getStats()[0]);
+            assertEquals(p1sCard, gameState.getCurrentCard());
+        }
+
+        /**
+         * Test createResponse when a User/Player requests to skip a turn which they can.
+         */
+        @Test
+        public void testCreateResponseSkipTurnValid () {
+
+        }
+
+        /**
+         * Test createResponse when a User requests to skip a turn, but they cannot.
+         */
+        @Test
+        public void testCreateResponseSkipTurnInvalid () {
+            interactor.createResponse(skipReq);
+            // Nothing should have changed. Therefore, assert game was how it was before the response.
+            assertEquals(p1, gameState.getCurrentPlayer());
+            assertFalse(gameState.getHasWinner());
+            assertEquals(firstCard, gameState.getCurrentCard());
+            for (Player p : game.getPlayers()) {
+                assertEquals(5, p.getNumCards());
+            }
+        }
+
+        /**
+         * Test createResponse when a User requests to pick up a Card and they can do so.
+         */
+        @Test
+        public void testCreateResponsePickUpCardValid () {
+            // Give a new bogus Hand to p1 so to validate p1 has no valid Cards.
+            Hand newHand = new Hand(new ArrayList<>());
+            Card bogus1 = new Card("Test", "17");
+            Card bogus2 = new Card("Test", "18");
+            newHand.addCard(bogus1);
+            newHand.addCard(bogus2);
+            p1.setHand(newHand);
+            // Have p1 pick up a Card. Only thing that should've changed is p1's number of Cards.
+            interactor.createResponse(pickUpReq);
+            assertEquals(p1, gameState.getCurrentPlayer());
+            assertFalse(gameState.getHasWinner());
+            assertEquals(firstCard, gameState.getCurrentCard());
+            for (Player p : game.getPlayers()) {
+                if(!(p.equals(p1))) {
+                    assertEquals(5, p.getNumCards());
+                } else {
+                    assertEquals(3, p1.getNumCards());
+                }
+            }
+        }
+
+        /**
+         * Test createResponse when a User requests to pick up a Card, but they cannot.
+         */
+        @Test
+        public void testCreateResponsePickUpCardInValid () {
+            PlayerGameResponseModel response = interactor.createResponse(pickUpReq);
+            // Nothing should have changed. Therefore, assert game was how it was before the response.
```

Comment:
Same as my above comment, is the user informed they cannot pick up a card? Maybe I missed the code where this happens.

Reply:
No they are not.